src/test/unit_tests/test_bucket_replication.js

```diff
@@ -798,29 +798,122 @@ async function put_object(s3_owner, bucket_name, key, optional_body) {
     console.log('put_object: ', util.inspect(res));
 }
 
-async function list_objects(s3_owner, bucket_name) {
-    const res = await s3_owner.listObjects({ Bucket: bucket_name }).promise();
-    console.log('list_objects: ', util.inspect(res));
-    return res;
-}
 
 async function delete_object(s3_owner, bucket_name, key) {
     const res = await s3_owner.deleteObject({ Bucket: bucket_name, Key: key }).promise();
     console.log('delete_object: ', util.inspect(res));
 }
+
+// The function lists *all* objects in a given bucket, including buckets that have more than 1k objs
+async function list_all_objs_in_bucket(s3owner, bucket, prefix) {
+    let isTruncated = true;
+    let marker;
+    const elements = [];
+    while (isTruncated) {
+      let params = { Bucket: bucket };
+      if (prefix) params.Prefix = prefix;
+      if (marker) params.Marker = marker;
```

Comment:
I think we will never enter this condition.
From what I understand: `marker` was initialized and not changed.